src/state/slices/opportunitiesSlice/opportunitiesSlice.ts

```diff
@@ -51,6 +58,23 @@ export type OpportunitiesState = {
 }
 
 export type OpportunityMetadataById = OpportunitiesState['lp' | 'staking']['byId']
+export type OpportunityDataById = OpportunitiesState['lp' | 'staking']['byAccountId']
+
+export type GetOpportunityMetadataInput = {
+  opportunityId: LpId | StakingId
+  opportunityType: 'lp' | 'staking'
+  defiType: DefiType
+}
+
+export type GetOpportunityDataInput = {
+  accountId: AccountId
+  opportunityId: LpId | StakingId
+  opportunityType: 'lp' | 'staking'
+  defiType: DefiType
+}
+
+type GetOpportunityMetadataOutput = any
+type GetOpportunityDataOutput = any
```

Comment:
Nice catch! This one was the opportunity to do a bit more cleanup:

- Types added and moved alongside all other types into a `./types` module to keep this one tidy and avoid circular deps
- While at it, renamed `metadata` to `byId` for consistency in `GetOpportunityUserDataOutput`
- Used those two output types in their matching opportunities reducers instead of repeating the payload typing in-place there

0ccb1d50e069f51d7a7a1914e27ab38795d1a389